### ISSUE TITLE
get_class on arrays of text

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/pynwb#529]
When an attibute is defined with `dtype='text'`, `get_class` always expects a `str`, even if `shape` is defined, e.g. `shape=(None,1)`. In that case it should take a list of strings.

```python
from pynwb.spec import NWBDatasetSpec, NWBNamespaceBuilder, NWBGroupSpec, \
    NWBAttributeSpec

name = 'test'
ns_path = name + '.namespace.yaml'
ext_source = name + '.extensions.yaml'

test_obj = NWBGroupSpec(
    neurodata_type_def='TestObj',
    doc='test object',
    datasets=[
        NWBDatasetSpec(
            name='filler_data', doc='data doc', dtype='int', shape=(None, 1),
            attributes=[
                NWBAttributeSpec(
                    name='text_data', dtype='text',
                    doc='text_data doc',
                    shape=(None, 1))
            ]
        )
    ],
    neurodata_type_inc='NWBDataInterface')

ns_builder = NWBNamespaceBuilder(name + ' extensions', name)
ns_builder.add_spec(ext_source, test_obj)
ns_builder.export(ns_path)



from pynwb import load_namespaces, get_class

# load custom classes
load_namespaces(ns_path)
TestObj = get_class('TestObj', name)
TestObj(source='source', name='name', filler_data=[1], text_data='hello')
```

runs without error, but if you replace the last line with:
```python

TestObj(source='source', name='name', filler_data=[1],
        text_data=['hello', 'goodbye'])
```

```python
Traceback (most recent call last):
  File "/Users/bendichter/dev/to_nwb/test_extension.py", line 36, in <module>
    text_data=['hello','goodbye'])
  File "/Users/bendichter/dev/pynwb/src/pynwb/form/utils.py", line 322, in func_call
    raise_from(TypeError(msg), None)
  File "<string>", line 3, in raise_from
TypeError: incorrect type for 'text_data' (got 'list', expected 'str')
```

**Comment by bendichter** [comment]
fixed on the entry point branch: 6a984671a7eab566d6311fec0d0e122aa1b88131

```python
from pynwb.spec import NWBDatasetSpec, NWBNamespaceBuilder, NWBGroupSpec, \
    NWBAttributeSpec

name = 'test'
ns_path = name + '.namespace.yaml'
ext_source = name + '.extensions.yaml'

test_obj = NWBGroupSpec(
    neurodata_type_def='TestObj',
    doc='test object',
    datasets=[
        NWBDatasetSpec(
            name='filler_data', doc='data doc', dtype='int', shape=(None, 1),
            attributes=[
                NWBAttributeSpec(
                    name='text_data', dtype='text',
                    doc='text_data doc',
                    shape=(None, 1))
            ]
        )
    ],
    neurodata_type_inc='NWBDataInterface')

ns_builder = NWBNamespaceBuilder(name + ' extensions', name)
ns_builder.add_spec(ext_source, test_obj)
ns_builder.export(ns_path)


from pynwb import load_namespaces
from pynwb.auto_class import get_class

# load custom classes
load_namespaces(ns_path)
TestObj = get_class(name, 'TestObj')
TestObj(source='source', name='name', filler_data=[1], text_data=['hello'])
```
works

**Comment by bendichter** [comment]
fixed
